ckan/plugins/core.py

```diff
@@ -240,10 +240,11 @@ def _get_service(plugin_name: str) -> Plugin:
     >>> assert isinstance(plugin, ActivityPlugin)
     """
     for group in GROUPS:
-        iterator = iter_entry_points(group=group,name=plugin_name)    # type: ignore
-        plugin = next(iterator, None)
-        if plugin:
-            return plugin.load()(name=plugin_name)
+        eps = entry_points(group=group, name=plugin_name)   # type:ignore
+        if len(eps.names):
+            plugin_ep = eps.pop(0)
+            return plugin_ep.load()(name=plugin_name)
```

Comment:
looking at the EntryPoints class definition treating it as a list seems to be deprecated. Something like this might be more future-proof:

```suggestion
        if ep:
            return ep[plugin_name].load()(name=plugin_name)
```